safe-global/safe-wallet-web
# fix: refetch recovery state via state changes

## What it solves

Resolves stale state when [cancelling recovery](https://www.notion.so/Delay-in-updating-Account-recovery-tx-after-it-was-cancelled-6edc1878f22d404e9830f9af7239cce7) or [updating the settings](https://www.notion.so/safe-global/A-manual-page-refresh-is-required-to-update-Account-recovery-settings-after-Edit-recovery-tx-executi-4b38fcfcfa37454bb5a375f17fac911c)

## How this PR fixes it

If the most recent transaction interacted with either a Delay Modifier or the MultiSend contract, the state is refetched. This has the benefit of not needing to fetch the transaction details from the gateway, which was sometimes throwing a 404 due to missing proposal and/or index.

## How to test it

1. Cancel a recovery and observe the transaction be removed from the pending recovery list.
2. Edit one/multiple settings of the recovery module and observe the settings update.

## Checklist
